### PR TITLE
[add] createMenuItem

### PR DESCRIPTION
**Pull request by yvesh**
Creates a Joomla menu item, for example:

$I->createMenuItem('My title', 'Weblinks', 'List All Web Link Categories', 'Main Menu');

### CODE REVIEW

File: src/JoomlaBrowser.php [hunk 1]
```diff
@@ -670,4 +670,47 @@ public function displayModuleOnAllPages($module)
...
+        $I->click(['xpath' =>  "//*[@id=\"menuList\"]/tbody/tr/td[2]/a[contains(text(), '" . $menu . "')]"]);;
+        $I->waitForText('Menus: Items', '30', ['css' => 'H1']);
+
+        $I->click(['xpath' => "//button[@onclick=\"Joomla.submitbutton('item.add')\"]"]);
+        $I->waitForText('Menus: New Item','30', ['css' => 'h1']);
+        $I->fillField(['id' => 'jform_title'], $menuTitle);
+
+        // Menu type (modal)
+        $I->click(['xpath' => "//a[@href=\"#menuTypeModal\"]"]);
+        $I->waitForElement('.iframe','30');
```
Comment (puneet0191) [inline src/JoomlaBrowser.php]: Can be replaced with a strict locator?

File: src/JoomlaBrowser.php [hunk 2]
```diff
@@ -670,4 +670,47 @@ public function displayModuleOnAllPages($module)
...
+     * @param   string  $menuCategory  - The category of the menu type (for example Weblinks)
+     * @param   string  $menuItem      - The menu item type / link text (for example List all Web Link Categories)
+     * @param   string  $menu          - The menu where the item should be created
+     */
+    public function createMenuItem($menuTitle, $menuCategory, $menuItem, $menu = "Main Menu")
+    {
+        $I = $this;
+        $I->amOnPage('administrator/index.php?option=com_menus&view=menus');
+        $I->waitForText('Menus', '30', ['css' => 'H1']);
+        $I->wait(1);
```
Comment (puneet0191) [inline src/JoomlaBrowser.php]: Why do we need to wait like this, you can use an element on the page for which you can wait to load compltely

**Comment by puneet0191** [comment]
Thanks, awesome work here :+1:

File: src/JoomlaBrowser.php (hunk 2 above)
Comment (yvesh) [inline src/JoomlaBrowser.php]: You are right. We don't have to wait - just copied the code from the old test. Works fine without it.

File: src/JoomlaBrowser.php [hunk 3]
```diff
@@ -670,4 +670,45 @@ public function displayModuleOnAllPages($module)
...
+     * @param   string  $menu          The menu where the item should be created
+     */
+    public function createMenuItem($menuTitle, $menuCategory, $menuItem, $menu = "Main Menu")
+    {
+        $I = $this;
+        $I->amOnPage('administrator/index.php?option=com_menus&view=menus');
+        $I->waitForText('Menus', '30', ['css' => 'H1']);
+
+        // Choose the right menu
+        $I->click(['xpath' =>  "//*[@id=\"menuList\"]/tbody/tr/td[2]/a[contains(text(), '" . $menu . "')]"]);;
```
Comment (javigomez) [inline src/JoomlaBrowser.php]: I have created this issue to improve this locator when we will have the chance: https://github.com/joomla-projects/joomla-browser/issues/62

**Comment by javigomez** [comment]
closes #28

**Comment by javigomez** [comment]
Tested. Works like a charm. Good job @yvesh 

Thanks

**Pull request merged**